qa-tests-backend/src/test/groovy/VulnReportingTest.groovy

```diff
@@ -0,0 +1,171 @@
+//                    .setNamespace(Constants.ORCHESTRATOR_NAMESPACE)
+//                    .setImage("quay.io/rhacs-eng/qa:nginx-1-9")
+//                    .addLabel("app", "nginx-test"),
+//            new Deployment()
+//                    .setName("nginx-deployment")
+//                    .setNamespace(SECONDARY_NAMESPACE)
+//                    .setImage("quay.io/rhacs-eng/qa:nginx-1-9")
+//                    .addLabel("app", "nginx-test"),
+    ]
+
+    @Shared
+    private MailServer mailServer
+
+    def setupSpec() {
+        mailServer = MailServer.createMailServer(orchestrator, true, false)
+        sleep 15 * 1000 // wait 15s for service to start
+
+        orchestrator.ensureNamespaceExists(SECONDARY_NAMESPACE)
+        orchestrator.batchCreateDeployments(DEPLOYMENTS)
+        DEPLOYMENTS.each { Services.waitForDeployment(it) }
+    }
+
+    def cleanupSpec() {
+        if (mailServer) {
+            mailServer.teardown(orchestrator)
+        }
+
+        DEPLOYMENTS.each { orchestrator.deleteDeployment(it) }
+        orchestrator.deleteNamespace(SECONDARY_NAMESPACE)
+    }
+
+    @Tag("BAT")
+    def "Verify vulnerability generated using a collection sends an email with a valid report attachment"() {
+        given:
+        "Collections feature is enabled and is central running on postgres"
+        Assume.assumeTrue(FeatureFlagService.isFeatureFlagEnabled("ROX_OBJECT_COLLECTIONS"))
+        Assume.assumeTrue(isPostgresRun())
+
+        and:
+        "a an email notifier is configured"
+        EmailNotifier notifier = new EmailNotifier("Vuln Reports Notifier",
+                mailServer.smtpUrl(),
+                true, true, NotifierOuterClass.Email.AuthMethod.DISABLED)
+        notifier.createNotifier()
+        assert notifier.id
+
+        and:
+        "a collection is created"
+        def collection = CollectionsService.createCollection(["struts-deployment"],
+                [Constants.ORCHESTRATOR_NAMESPACE])
+        assert collection.id
+
+        and:
+        "a report is configured"
+        def report = VulnReportService.createVulnReportConfig(collection.id, notifier.id)
+        assert report.id
+
+        when:
+        "a report is generated"
+        assert VulnReportService.runReport(report.id)
+
+        then:
+        "the email server should've gotten an email with the report"
+
+        def emails = mailServer.findEmailsByToEmail(Constants.EMAIL_NOTIFER_SENDER)
+        assert emails.size() == 1
+
+        def email = emails[0]
+        def emailId = (String) email["id"]
+
+        assert emailId
+        assert email["subject"] =~ /(StackRox|RHACS) Image Vulnerability Report for (\d+)-(.*)-(\d+)/
+        assert email["html"] =~ /has found vulnerabilities/
+
+        // Since this is a BAT test, keep it simple and only validate we got the attachment and it's not 0 bytes
+        Object[] attachments = email["attachments"]
+        assert attachments.size() >= 2
```

Comment:
Guessing this is 2 because of the embedded logo which shows up as attachment? Comment would be nice

Reply:
yep. Done